cmd/gateway-main.go

```diff
@@ -113,6 +113,8 @@ ENVIRONMENT VARIABLES:
 
 EXAMPLES:
   1. Start minio gateway server for GCS backend.
+      $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
```

Comment:
@vadmeste mentioning that this is optional will be confusing because we will be saying authentication to google itself will be optional.

There are different ways to provide auth info to gcs-sdk. It would be too much info to explain all the different ways here. So best thing would be to put GOOGLE_APPLICATION_CREDENTIALS in this help message as it is the best way for minio deployements. And then point to google documentation for further info.